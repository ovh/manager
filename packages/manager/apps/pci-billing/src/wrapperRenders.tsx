import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import {
  ShellContext,
  ShellContextType,
} from '@ovh-ux/manager-react-shell-client';
import { vi } from 'vitest';

export const shellContext = {
  environment: {
    getUser: () => ({ ovhSubsidiary: 'FR', currency: 'â‚¬' }),
  },
  shell: {
    navigation: {
      getURL: vi.fn(),
    },
  },
};

const queryClient = new QueryClient({
  defaultOptions: {
    mutations: {
      retry: false,
    },
    queries: {
      retry: false,
    },
  },
});

export const wrapper = ({ children }) => (
  <QueryClientProvider client={queryClient}>
    <ShellContext.Provider
      value={(shellContext as unknown) as ShellContextType}
    >
      {children}
    </ShellContext.Provider>
  </QueryClientProvider>
);
