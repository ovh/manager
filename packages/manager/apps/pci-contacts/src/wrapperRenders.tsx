import {
  ShellContext,
  ShellContextType,
} from '@ovh-ux/manager-react-shell-client';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { ReactNode } from 'react';
import { MemoryRouter } from 'react-router-dom';
import { vi } from 'vitest';

export const shellContext = ({
  environment: {
    getUser: () => ({ ovhSubsidiary: 'FR' }),
    getRegion: () => 'EU',
  },
  shell: {
    navigation: {
      navigateTo: vi.fn(),
      getURL: vi.fn().mockImplementation((app, path) => {
        if (app === 'dedicated' && path === '#/billing/history') {
          return Promise.resolve('https://billing-url');
        }
        return Promise.resolve('');
      }),
    },
  },
} as unknown) as ShellContextType;

export const createWrapper = (
  contextValue: ShellContextType = shellContext,
) => {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        staleTime: 300_000,
      },
    },
  });

  const Wrapper = ({ children }: { children: ReactNode }) => (
    <QueryClientProvider client={queryClient}>
      <ShellContext.Provider value={contextValue}>
        <MemoryRouter>{children}</MemoryRouter>
      </ShellContext.Provider>
    </QueryClientProvider>
  );
  return Wrapper;
};
