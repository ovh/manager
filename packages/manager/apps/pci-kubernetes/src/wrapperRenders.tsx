import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import {
  ShellContext,
  ShellContextType,
} from '@ovh-ux/manager-react-shell-client';
import { vi } from 'vitest';

export const shellContext = {
  environment: {
    getRegion: vi.fn().mockReturnValue('EU'),
    getUser: () => ({
      ovhSubsidiary: 'mocked_ovhSubsidiary',
      currency: { code: 'EUR' },
    }),
  },
  shell: {
    navigation: {
      getURL: vi.fn(),
    },
  },
};

const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
    },
  },
});

export const wrapper = ({ children }) => (
  <QueryClientProvider client={queryClient}>
    <ShellContext.Provider
      value={(shellContext as unknown) as ShellContextType}
    >
      {children}
    </ShellContext.Provider>
  </QueryClientProvider>
);
