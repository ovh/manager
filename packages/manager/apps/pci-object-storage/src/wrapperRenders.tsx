import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import {
  ShellContext,
  ShellContextType,
} from '@ovh-ux/manager-react-shell-client';
import { vi } from 'vitest';
import { createMemoryRouter, RouterProvider } from 'react-router-dom';
import React from 'react';

export const shellContext = {
  environment: {
    getUser: () => ({ ovhSubsidiary: 'mocked_ovhSubsidiary' }),
  },
  shell: {
    navigation: {
      getURL: vi.fn(),
    },
    tracking: {
      trackPage: vi.fn(),
      trackClick: vi.fn(),
    },
  },
};

const queryClient = new QueryClient({
  defaultOptions: {
    mutations: {
      retry: false,
    },
    queries: {
      retry: false,
    },
  },
});

export const wrapper = ({ children }) => (
  <QueryClientProvider client={queryClient}>
    <ShellContext.Provider
      value={(shellContext as unknown) as ShellContextType}
    >
      {children}
    </ShellContext.Provider>
  </QueryClientProvider>
);

export const mockShellContext = {
  environment: {
    getRegion: vi.fn().mockReturnValue('EU'),
    getUser: vi.fn().mockReturnValue({ ovhSubsidiary: 'mocked_ovhSubsidiary' }),
    getApplication: vi.fn(),
  },
  shell: {
    environment: {
      getRegion: vi.fn().mockReturnValue('EU'),
      getUser: vi.fn().mockReturnValue({}),
      getApplication: vi.fn(),
    },
    navigation: {
      getURL: vi.fn(),
    },
    tracking: {
      trackPage: vi.fn(),
      trackClick: vi.fn(),
    },
  },
};

export const wrapperShow = ({ children }: { children: React.ReactNode }) => {
  const router = createMemoryRouter(
    [
      {
        path: '/',
        element: children,
        handle: {
          crumb: () => 'Home',
        },
      },
    ],
    {
      initialEntries: ['/'],
      future: {
        v7_startTransition: true,
        v7_relativeSplatPath: true,
      } as any,
    },
  );

  return (
    <QueryClientProvider client={queryClient}>
      <ShellContext.Provider value={mockShellContext as any}>
        <RouterProvider router={router} />
      </ShellContext.Provider>
    </QueryClientProvider>
  );
};

export const wrapperOffsiteReplication = ({
  children,
}: {
  children: React.ReactNode;
}) => (
  <QueryClientProvider client={queryClient}>
    <ShellContext.Provider value={mockShellContext as any}>
      {children}
    </ShellContext.Provider>
  </QueryClientProvider>
);
