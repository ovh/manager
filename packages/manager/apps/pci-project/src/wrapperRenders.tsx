import {
  ShellContext,
  ShellContextType,
} from '@ovh-ux/manager-react-shell-client';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { ReactNode } from 'react';
import { MemoryRouter } from 'react-router-dom';
import { vi } from 'vitest';

export const shellContext = ({
  environment: {
    getUser: () => ({ ovhSubsidiary: 'FR' }),
    getRegion: () => 'EU',
  },
  shell: {
    navigation: {
      navigateTo: vi.fn(),
      getURL: vi.fn().mockImplementation((app, path) => {
        if (app === 'dedicated' && path === '#/billing/history') {
          return Promise.resolve('https://billing-url');
        }
        if (app === 'dedicated' && path === '#/contacts/services') {
          return Promise.resolve('https://contacts-url');
        }
        return Promise.resolve('https://mocked-url');
      }),
    },
  },
} as unknown) as ShellContextType;

export const createWrapper = (
  contextValue: ShellContextType = shellContext,
) => {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        staleTime: 300_000,
      },
    },
  });

  const Wrapper = ({ children }: { children: ReactNode }) => (
    <QueryClientProvider client={queryClient}>
      <ShellContext.Provider value={contextValue}>
        <MemoryRouter>{children}</MemoryRouter>
      </ShellContext.Provider>
    </QueryClientProvider>
  );
  return Wrapper;
};
