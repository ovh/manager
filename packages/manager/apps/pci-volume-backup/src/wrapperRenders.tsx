import { ReactNode } from 'react';
import { QueryClientProvider } from '@tanstack/react-query';
import {
  ShellContext,
  ShellContextType,
} from '@ovh-ux/manager-react-shell-client';
import { vi } from 'vitest';
import { ErrorBoundary } from 'react-error-boundary';
import queryClient from './queryClient';

export const shellContext = ({
  environment: {
    getUser: () => ({ ovhSubsidiary: 'mocked_ovhSubsidiary' }),
  },
  shell: {
    navigation: {
      navigateTo: vi.fn(),
      getURL: vi.fn(),
    },
  },
} as unknown) as ShellContextType;

export const createWrapper = () => {
  const Wrapper = ({ children }: { children: ReactNode }) => (
    <QueryClientProvider client={queryClient}>
      <ShellContext.Provider value={shellContext}>
        <ErrorBoundary fallbackRender={() => <div>error boundary</div>}>
          {children}
        </ErrorBoundary>
      </ShellContext.Provider>
    </QueryClientProvider>
  );
  return Wrapper;
};
