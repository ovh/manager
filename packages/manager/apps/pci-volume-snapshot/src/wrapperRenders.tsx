import { ReactNode } from 'react';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import {
  ShellContext,
  ShellContextType,
} from '@ovh-ux/manager-react-shell-client';
import { vi } from 'vitest';
import { ErrorBoundary } from 'react-error-boundary';

export const shellContext = {
  environment: {
    getUser: () => ({ ovhSubsidiary: 'mocked_ovhSubsidiary' }),
  },
  shell: {
    navigation: {
      getURL: vi.fn(),
    },
  },
};

export const createWrapper = () => {
  const queryClient = new QueryClient({
    defaultOptions: {
      mutations: {
        retry: false,
      },
      queries: {
        retry: false,
      },
    },
  });
  const Wrapper = ({ children }: { children: ReactNode }) => (
    <QueryClientProvider client={queryClient}>
      <ShellContext.Provider
        value={(shellContext as unknown) as ShellContextType}
      >
        <ErrorBoundary fallbackRender={() => <div>error boundary</div>}>
          {children}
        </ErrorBoundary>
      </ShellContext.Provider>
    </QueryClientProvider>
  );
  return Wrapper;
};
