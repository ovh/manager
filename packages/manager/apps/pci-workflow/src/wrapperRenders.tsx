import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

const queryClient = new QueryClient();
export const wrapper = ({ children }) => (
  <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
);
