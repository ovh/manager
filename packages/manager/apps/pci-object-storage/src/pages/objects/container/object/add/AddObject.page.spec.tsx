import { render } from '@testing-library/react';
import AddObjectPage from './AddObject.page';
import { wrapper } from '@/wrapperRenders';

describe('AddObjectPage', () => {
  it('should display correctly', () => {
    const { container } = render(<AddObjectPage />, { wrapper });
    expect(container).toMatchSnapshot();
  });
});
