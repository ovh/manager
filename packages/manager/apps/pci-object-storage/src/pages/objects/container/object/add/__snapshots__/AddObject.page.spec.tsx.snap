// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AddObjectPage > should display correctly 1`] = `
<div>
  <ods-modal
    color="information"
    data-testid="pciModal-modal"
    is-open="true"
  >
    <ods-text
      preset="heading-3"
    >
      pci_projects_project_storages_containers_container_object_add_object_title
    </ods-text>
    <slot
      name="content"
    >
      <ods-spinner
        class="block text-center mt-6 mb-8"
        data-testid="pciModal-spinner"
        size="md"
      />
    </slot>
    <ods-button
      color="primary"
      data-testid="pciModal-button_cancel"
      is-disabled="true"
      label="pci_projects_project_storages_containers_container_object_add_cancel_label"
      slot="actions"
      variant="ghost"
    />
    <ods-button
      color="primary"
      data-testid="pciModal-button_submit"
      is-disabled="true"
      label="pci_projects_project_storages_containers_container_object_add_submit_label"
      slot="actions"
    />
  </ods-modal>
</div>
`;
